dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java

```diff
@@ -383,77 +377,36 @@ public Map<String, Object> updateUser(User loginUser, int userId,
                                           String phone,
                                           String queue,
                                           int state,
-                                          String timeZone) throws IOException {
+                                          String timeZone) {
         Map<String, Object> result = new HashMap<>();
         result.put(Constants.STATUS, false);
 
         if (resourcePermissionCheckService.functionDisabled()) {
-            putMsg(result, Status.FUNCTION_DISABLED);
-            return result;
+            throw new ServiceException(Status.FUNCTION_DISABLED.getMsg());
         }
         if (check(result, !canOperator(loginUser, userId), Status.USER_NO_OPERATION_PERM)) {
-            return result;
+            throw new ServiceException(Status.USER_NO_OPERATION_PERM.getMsg());
```

Comment:
I preview reviewed have a mistake, you can directly use `throw new ServiceException(Status.USER_NO_OPERATION_PERM);`